bindata/network/ovn-kubernetes/ovnkube-master.yaml

```diff
@@ -268,6 +329,61 @@ spec:
                   sleep 2
                   done
                 fi
+
+                election_timer="${OVN_SB_RAFT_ELECTION_TIMER}"
+                echo "Setting sb-db raft election timer to ${election_timer} ms"
+                retries=0
+                while current_election_timer=$(ovs-appctl -t /var/run/ovn/ovnsb_db.ctl cluster/status OVN_Southbound 2>/dev/null \
+                  | grep "Election" | sed "s/.*:[[:space:]]//"); do
+                  if [[ -z "${current_election_timer}" ]]; then
+                    (( retries += 1 ))
+                    if [[ "${retries}" -gt 10 ]]; then
+                      echo "Failed to get current sb-db raft election timer value after multiple attempts. Exiting..."
+                      exit 1
+                    fi
+                    sleep 2
+                  else
+                    break
+                  fi
+                done
+
+                retries=0
+                while is_candidate=$(ovs-appctl -t /var/run/ovn/ovnsb_db.ctl cluster/status OVN_Southbound 2>/dev/null \
+                  | grep "Role: candidate" ); do
+                  if [[ ! -z "${is_candidate}" ]]; then
+                    (( retries += 1 ))
+                    if [[ "${retries}" -gt 10 ]]; then
+                      echo "Cluster node (sb-db raft) is in candidate role for prolonged time. Continuing..."
+                    fi
+                    sleep 2
+                  else
+                    break
+                  fi
+                done
+
+                is_leader=$(ovs-appctl -t /var/run/ovn/ovnsb_db.ctl cluster/status OVN_Southbound 2>/dev/null \
+                  | grep "Role: leader")
+                if [[ ! -z "${is_leader}" ]]; then
+                  while [[ ${current_election_timer} != ${election_timer} ]]; do
+                    max_electinon_timer=$((${current_election_timer} * 2))
+                    if [[ ${election_timer} -le ${max_electinon_timer} ]]; then
+                      ovs-appctl -t /var/run/ovn/ovnsb_db.ctl cluster/change-election-timer OVN_Southbound ${election_timer}
+                      if [[ $? != 0 ]]; then
```

Comment:
`if` can check exit status directly without having to use `$?` , and we can use `!` to check for failure.

```bash
if ! ovs-appctl -t /var/run/ovn/ovnsb_db.ctl cluster/change-election-timer OVN_Southbound ${election_timer}; then
    echo "Failed to set sb-db raft election timer ${election_timer}. Exiting..."
    exit 2
fi
```

Reply:
I am not an scripting expert but will it log the error message from ovs-appctl in case of failure, if it's used in if statement ?

Reply:
yes it will display output, stdout and stderr are the same.

```bash
$ if ! ls /tmp/nosuch ; then echo yes ; fi
ls: cannot access '/tmp/nosuch': No such file or directory
yes
```

Reply:
fixed.